applications/serial_lte_modem/src/slm_ppp.c

```diff
@@ -200,6 +200,30 @@ static int ppp_start_failure(int ret)
 	return ret;
 }
 
+static unsigned int retrieve_mtu(void)
+{
+	struct pdn_context_dynamic_info populated_info = { 0 };
+
+	if (!pdn_context_dynamic_params_get(PDP_CID, &populated_info)) {
```

Comment:
Not a strict requirement, though I think it is a good practice to keep the function calls outside the if statement and keep the happy-path unindented, so instead of
```
if (!func()) {
    ...happy path...
    return;
}
...error handling...
```
we do
```
err = func();
if (err) {
    LOG_ERR("func() failed, err %d", err); /* could be LOG_DBG etc. */
    ...error handling.... 
    return;
}
...happy path...
```

Reply:
Actually this block:
```
	LOG_DBG("Could not retrieve MTU, using fallback value.");
	BUILD_ASSERT(sizeof(ppp_data_buf) >= CONFIG_SLM_PPP_FALLBACK_MTU);
	return CONFIG_SLM_PPP_FALLBACK_MTU;
```
needs to be excuted either on error case or the case that ISP doesn't send any MTU for either IPv4 or IPv6 which is really nor an error case. It was the one of the reason I wanted to implement like that. Another reason was preventing code duplication even if it was seperated as a function.

Reply:
Yes, that block would have to be both in the error handling and at the end. It is ok to leave as is.